src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/support/SupportController.java

```diff
@@ -98,6 +106,42 @@ public void resendStaffNotifications(
         }
     }
 
+    @PutMapping("/claim/{referenceNumber}/claimDocumentType/{claimDocumentType}/uploadDocumentToDocumentManagement")
+    @ApiOperation("Upload document to document Management")
+    public void uploadDocumentToDocumentManagement(
+        @PathVariable("referenceNumber") String referenceNumber,
+        @PathVariable("claimDocumentType") ClaimDocumentType claimDocumentType,
+        @RequestHeader(value = HttpHeaders.AUTHORIZATION, required = false) String authorisation) {
+
+        Claim claim = claimService.getClaimByReferenceAnonymous(referenceNumber)
+            .orElseThrow(() -> new NotFoundException(String.format(CLAIM_DOES_NOT_EXIST, referenceNumber)));
+        if (StringUtils.isBlank(authorisation)) {
+            throw new BadRequestException(AUTHORISATION_IS_REQUIRED);
+        }
+        switch (claimDocumentType) {
+            case SEALED_CLAIM:
+                documentsService.generateSealedClaim(claim.getExternalId(), authorisation);
+                break;
+            case CLAIM_ISSUE_RECEIPT:
+                documentsService.generateClaimIssueReceipt(claim.getExternalId(), authorisation);
+                break;
+            case DEFENDANT_RESPONSE_RECEIPT:
+                documentsService.generateDefendantResponseReceipt(claim.getExternalId(), authorisation);
+                break;
+            case CCJ_REQUEST:
+                documentsService.generateCountyCourtJudgement(claim.getExternalId(), authorisation);
+                break;
+            case SETTLEMENT_AGREEMENT:
+                documentsService.generateSettlementAgreement(claim.getExternalId(), authorisation);
+                break;
+            default:
+                throw new BadRequestException("ClaimDocumentType " + claimDocumentType + " is not supported");
+        }
+        claimService.getClaimByReferenceAnonymous(referenceNumber)
+            .ifPresent(updatedClaim -> updatedClaim.getClaimDocument(claimDocumentType)
+                .orElseThrow(() -> new NotFoundException("Unable to upload the document. Please try again later")));
+    }
+
     @PutMapping("/claim/resend-rpa-notifications")
     @ApiOperation("Resend notifications for multiple citizen claims")
     public void resendRPANotifications(
```

Comment:
not your change, but can you check if authorisation can be optional. rather change it as required = true

Reply:
don't think its optional ..don't know why it is that way.